models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php

```diff
@@ -774,8 +774,9 @@ public function getColumnKeys()
 
     /**
      * @param DataObject\ClassDefinition $class
+     * @param array $params
      */
-    public function classSaved($class)
+    public function classSaved($class, $params = [])
```

Comment:
```suggestion
    public function classSaved($class/**, $params = [] **/)
```
We should keep the definition commented out and introduce it in v11. 
For compatibility we can use `func_get_args()` to fill `$params` in v10.6.

Reply:
But then we can not use the interface and get deprecation warnings with this data type, right?

Reply:
The interface requires the `$params` variable

Reply:
Why not? I mean the interface should also have the same signature (with the commented `$params`)  - then it shouldn't be a problem. Then in v11, we change both the interface and the implementations.

Reply:
But then the interface is not compatible with the other field types, or can a child method have more parameters than the interface?

Reply:
I think we can make the interface without methods in 10.6 (only a `@method` annotation). This should work.

Reply:
@blankse right, this is what I meant 😉

Reply:
> But then the interface is not compatible with the other field types, or can a child method have more parameters than the interface?

Yes, that's possible, if the additional parameters are at the end and have a default value. See: https://3v4l.org/F2hKa

Reply:
None of our classes (that implement the interface) really uses the `$param` parameter - a custom implementation must therefore use the default value or don't use the `$params` parameter at all ....